vault/core.go

```diff
@@ -2816,6 +2816,19 @@ func (c *Core) SetLogLevel(level log.Level) {
 	}
 }
 
+func (c *Core) SetLogLevelByName(name string, level log.Level) error {
+	c.allLoggersLock.RLock()
+	defer c.allLoggersLock.RUnlock()
+	for _, logger := range c.allLoggers {
+		if logger.Name() == name {
```

Comment:
Equality, or prefix match (only from dot delimiter boundaries)?

Reply:
My thinking is that if I want to turn up logging for "core.identity", I shouldn't have to know that there may be subsystems within identity that create subloggers (e.g. "core.identity.storagepacker") that I probably also want turned up.  I may not even be aware they exist based on the log level I usually use.  Maybe we could take a regexp?

Reply:
That's a good question. I think that supporting equality would provide a better UX. One sub logger might be far noisier than the one the user is intending to modify. If we get feedback that folks want the ability to override an entire class of loggers then we could add an optional API param to allow for it.

Reply:
I had that hesitation too, which was why I suggested regexps, that way both options would be available.

Reply:
I'm probably trying to make it do too much, let's go with your simpler approach for now.

Reply:
This actually made me notice something, @ncabatoff. It looks like there are numerous sub-loggers that are not accounted for in `allLoggers` by calling `AddLogger`. I'm going to go through and audit calls to `logger.Named` and call `AddLogger` to ensure that these new API endpoints can actually modify all the logger names that users will see in their logs.

Reply:
There are some loggers that are short-lived (e.g. for initialization of a backend) that I'll leave out of `allLoggers`.